libs/journeys/ui/src/components/StepHeader/PaginationBullets/PaginationBullets.tsx

```diff
@@ -70,3 +97,23 @@ export function PaginationBullets(): ReactElement {
     </Box>
   )
 }
+
+function removeAlreadyVisitedBlocksFromHistory(
+  blockHistory: TreeBlock[],
+  currentActiveBlock: TreeBlock<StepFields>
+): void {
+  // paginate to beginning if you are back at the starting point
+  if (currentActiveBlock?.id === blockHistory[0]?.id) {
+    blockHistory.splice(1, blockHistory.length - 1)
+  }
+  // paginate back if you have already visited card
+  let prevCards: TreeBlock[] = []
+  prevCards = blockHistory.filter((block) => block.id === currentActiveBlock.id)
+  if (prevCards.length > 1) {
+    blockHistory.forEach((block, i) => {
+      if (block.id === currentActiveBlock.id) {
+        blockHistory.splice(i + 1, blockHistory.length - i)
+      }
+    })
+  }
```

Comment:
I think creating this array is redundant. You will always need to loop through all of blockHistory, except you need to do it twice if `prevCards.length > 0`

```suggestion
    blockHistory.forEach((block, i) => {
      if (block.id === currentActiveBlock.id) {
        blockHistory.splice(i + 1, blockHistory.length - i)
      }
    })
```